ivy_tests/test_ivy/test_functional/test_core/test_creation.py

```diff
@@ -208,11 +208,33 @@ def _asarray_helper(draw):
             shared_dtype=True,
         )
     )
+    x_list = ivy.nested_map(x, lambda x: x.tolist(), shallow=False)
+    sh = draw(helpers.get_shape(min_num_dims=1))
+    sh = ivy.Shape(sh)
+    np_array = x[0]
+    ivy_array = ivy.array(x[0], dtype=x_dtype[0])
```

Comment:
this caused problems because it's created in a backend that is different that the one being tested, will creating it inside the test function body before calling `helpers.test_function` help create it in the correct backend? and even so, the input will still be evaluated in both test and groundtruth backends so I don't know if dynamic backend will help in this case